### PR TITLE
feat: add write_source_file to public API

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

**Review by alexeagle — COMMENTED** [review]
Why is it better than one rule with an attribute which is a singleton list?

**Comment by gregmagolan** [comment]
> Why is it better than one rule with an attribute which is a singleton list?

For the single file case I think it reads better, especially in the `executable = True` case. This is more similar to the `write_file` rule from skylib too so if users are familiar with that they will find this for recognizable:

```
write_source_file(
    name = "write_a",
    in_file = "a_desired,
    out_file = "a",
    executable = True,
)
```

vs

```
    write_source_files(
        name = "write_a",
        files = {
            "a": "a_desired",
        },
        executable = True,
    )
```

File: lib/write_source_files.bzl [hunk 1]
```diff
@@ -113,9 +110,9 @@ def write_source_files(
 
             To set different executable permissions on different source tree files use multiple `write_source_files` targets.
 
-        additional_update_targets: List of other `write_source_files` or other executable updater targets to call in the same run.
```
Comment (jbedard) [inline lib/write_source_files.bzl:116]: are "other executable updater targets" not supported?

**Review by jbedard — APPROVED** [review]

File: lib/write_source_files.bzl (hunk 1 above)
Comment (gregmagolan) [reply lib/write_source_files.bzl:116]: they are but only with an undocumented provider so easier to remove this line than to document the provider

**Pull request merged**